scripts/inline_comments_data_collection.py

```diff
@@ -0,0 +1,181 @@
+import json
+import logging
+import os
+
+from libmozdata.phabricator import PhabricatorAPI
+
+from bugbug.tools.code_review import PhabricatorReviewData
+from bugbug.utils import get_secret
+
+review_data = PhabricatorReviewData()
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+os.makedirs("patches", exist_ok=True)
+os.makedirs("comments", exist_ok=True)
+os.makedirs("dataset", exist_ok=True)
+
+
+class NoDiffsFoundException(Exception):
+    def __init__(self, patch_id):
+        super().__init__(f"No diffs found for the given patch ID: {patch_id}")
+        self.patch_id = patch_id
+
+
+class NoTransactionsFoundException(Exception):
+    def __init__(self, patch_id):
+        super().__init__(f"No transactions found for the given patch ID: {patch_id}")
+        self.patch_id = patch_id
+
+
+class NoDiffFoundForPHIDException(Exception):
+    def __init__(self, phid):
+        super().__init__(f"No diff found for PHID {phid}")
+        self.phid = phid
+
+
+class NoRevisionFoundException(Exception):
+    def __init__(self, phid):
+        super().__init__(f"No revision found for the given revision PHID: {phid}")
+        self.phid = phid
+
+
+class PhabricatorClient:
+    def __init__(self):
+        self.api = PhabricatorAPI(get_secret("PHABRICATOR_TOKEN"))
+
+    def find_revision_from_patch(self, patch_id):
+        diffs = self.api.search_diffs(diff_id=patch_id)
+
+        if not diffs:
+            raise NoDiffsFoundException(patch_id)
+
+        revision_phid = diffs[0]["revisionPHID"]
+        return revision_phid
+
+    def find_transactions_from_patch(self, patch_id):
+        revision_phid = self.find_revision_from_patch(patch_id)
+        transactions = self.api.request(
+            "transaction.search", objectIdentifier=revision_phid
+        )["data"]
+
+        if not transactions:
+            raise NoTransactionsFoundException(patch_id)
+
+        return transactions
+
+    def get_diff_info_from_phid(self, phid):
+        diffs = self.api.search_diffs(diff_phid=phid)
+        if not diffs:
+            raise NoDiffFoundForPHIDException(phid)
+        return diffs[0]["id"], diffs[0]["revisionPHID"]
+
+    def find_bugid_from_revision_phid(self, phid):
+        revision = self.api.load_revision(rev_phid=phid)
+        if not revision:
+            raise NoRevisionFoundException(phid)
```

Comment:
I do not think that you need to manually raise an error here. If there is no data found, the `load_revision()` method will raise `PhabricatorRevisionNotFoundException`.